app/src/main/java/com/cornellappdev/android/eatery/model/EateryModel.java

```diff
@@ -0,0 +1,234 @@
+  protected String mName;
+  protected String mNickName;
+  private String mCloseTime;
+  protected CampusArea mArea;
+  private boolean mMatchesFilter = true;
+  private boolean mMatchesSearch = true;
+  private String mBuildingLocation;
+  protected List<String> mPayMethods;
+  private List<String> mSearchedItems;
+  private boolean openPastMidnight;
+
+  public List<String> getSearchedItems() {
+    return mSearchedItems;
+  }
+
+  public void setSearchedItems(List<String> searchedItems) {
+    this.mSearchedItems = searchedItems;
+  }
+
+  public abstract ZonedDateTime getNextOpening();
+
+  public abstract ZonedDateTime getCloseTime();
+
+  public String stringTo() {
+    String info = "Name/mNickName: " + mName + "/" + mNickName;
+    String locationString = "Location: " + ", Area: " + mArea;
+    String payMethodsString = "Pay Methods: " + mPayMethods.toString();
+    String menuString = "";
+
+    return info + "\n" + locationString + "\n" + payMethodsString + "\n" + "Menu" + "\n"
+        + menuString;
+  }
+
+  public abstract List<String> getMealItems();
+
+  public boolean isOpen() {
+    return getCurrentStatus() == Status.OPEN;
+  }
+
+  public abstract Status getCurrentStatus();
+
+  public String getName() {
+    return mName;
+  }
+
+  public void setName(String name) {
+    this.mName = name;
+  }
+
+  public String getNickName() {
+    return mNickName;
+  }
+
+  public void setNickName(String nickName) {
+    this.mNickName = nickName;
+  }
+
+  public boolean matchesFilter() {
+    return mMatchesFilter;
+  }
+
+  public void setMatchesFilter(boolean yesOrNo) {
+    mMatchesFilter = yesOrNo;
+  }
+
+  public boolean matchesSearch() {
+    return mMatchesSearch;
+  }
+
+  public void setMatchesSearch(boolean yesOrNo) {
+    mMatchesSearch = yesOrNo;
+  }
+
+  public List<String> getPayMethods() {
+    return mPayMethods;
+  }
+
+  public void setPayMethods(List<String> paymentMethods) {
+    this.mPayMethods = paymentMethods;
+  }
+
+  public String getBuildingLocation() {
+    return mBuildingLocation;
+  }
+
+  public void setBuildingLocation(String buildingLocation) {
+    this.mBuildingLocation = buildingLocation;
+  }
+
+  public int getId() {
+    return id;
+  }
+
+  public void setId(int id) {
+    this.id = id;
+  }
+
+  public boolean isHardCoded() {
+    return isHardCoded;
+  }
+
+  public void setHardCoded(boolean hardCoded) {
+    isHardCoded = hardCoded;
+  }
+
+  public LatLng getLatLng() {
+    return new LatLng(lat, lng);
+  }
+
+  public void setLatLng(LatLng latlng) {
+    this.lat = latlng.latitude;
+    this.lng = latlng.longitude;
+  }
+
+  public boolean isOpenPastMidnight() {
+    return openPastMidnight;
+  }
+
+  public void setOpenPastMidnight(boolean openPastMidnight) {
+    this.openPastMidnight = openPastMidnight;
+  }
+
+  public CampusArea getArea() {
+    return mArea;
+  }
+
+  public void setArea(CampusArea area) {
+    this.mArea = area;
+  }
+
+
+  public enum Status {
```

Comment:
can this be moved into its own file too

Reply:
That can be, I believe that is a relic of the past structure.

Reply:
Can we hold off on this until after this merge, that way it doesn't completely break future merges (status is used a lot, and it'll need a codebase-wide import refactoring)